aws/rust-runtime/aws-runtime/src/auth/sigv4.rs

```diff
@@ -76,12 +75,13 @@ impl SigV4Signer {
         super::settings(operation_config)
     }
 
-    fn signing_params<'a>(
+    /// Creates a [`v4::SigningParams`] from the given arguments.
+    pub fn signing_params<'a>(
         settings: SigningSettings,
         identity: &'a Identity,
         operation_config: &'a SigV4OperationSigningConfig,
         request_timestamp: SystemTime,
-    ) -> Result<v4::SigningParams<'a, SigningSettings>, SigV4SigningError> {
+    ) -> Result<v4::SigningParams<'a, SigningSettings>, BoxError> {
```

Comment:
I should have caught this in earlier reviews, but I think we should instead make `SigV4SigningError` an opaque/informative public error so that we can add methods to it in the future if we need to. See https://github.com/smithy-lang/smithy-rs/blob/main/design/src/rfcs/rfc0022_error_context_and_compatibility.md

It's pretty easy to do this. Just rename the existing error enum to "ErrorKind", and make a public struct with the previous name.